src/parser.js

```diff
@@ -0,0 +1,126 @@
+const parser = require("@babel/parser");
+const traverse = require("@babel/traverse").default;
+const EventEmitter = require("events").EventEmitter;
+
+function createProp(attributeNode) {
+  return getPropName(attributeNode);
+
+  function getPropName(attributeNode) {
+    switch (attributeNode.type) {
+      case "JSXAttribute":
+        return attributeNode.name.name;
+      case "JSXSpreadAttribute":
+        return "{...}";
+      default:
+        throw new Error(`unexpected node type: ${attributeNode.type}`);
+    }
+  }
+}
+
+function createComponent(componentNode) {
+  return getDottedName(componentNode.openingElement.name);
+
+  function getDottedName(nameNode) {
+    switch (nameNode.type) {
+      case "JSXMemberExpression":
+        return [nameNode.object, nameNode.property]
+          .map(getDottedName)
+          .join(".");
+      case "JSXIdentifier":
+        return nameNode.name;
+      default:
+        throw new Error(`unexpected node type: ${nameNode.type}`);
+    }
+  }
+}
```

Comment:
Can you pull `getPropName` and `getDottedName` out? I think it's confusing defining functions after `return` and they don't need to be defined inside of those functions anyway.

Reply:
I was having an idea of not just passing the name string of the founded thing (`component`,`prop`,`child`), but an object structure of useful info about it, like for example the sibling props of a `prop`, useful to generate a report about what props are always used together in order to refactor them in one data structure, we'll return to this function structure later if you want, I'll pass strings for now👍

Reply:
I think passing an object is good! I just would prefer the functions like this:

```js
function createComponent(componentNode) {
  return getDottedName(componentNode.openingElement.name);
}

function getDottedName(nameNode) {
  switch (nameNode.type) {
    case "JSXMemberExpression":
      return [nameNode.object, nameNode.property].map(getDottedName).join(".");
    case "JSXIdentifier":
      return nameNode.name;
    default:
      throw new Error(`unexpected node type: ${nameNode.type}`);
  }
}
```

Instead of inside each other

Reply:
`getDottedName` used only in `createComponent`, so is `getAttributeName` we should not call them directly i think.